conda/core/linked_data.py

```diff
@@ -105,6 +105,18 @@ def all_subdir_urls(self):
                 subdir_urls.add(subdir_url)
         return subdir_urls
 
+    def query(self, package_ref_or_match_spec):
+        # returns a generator
+        param = package_ref_or_match_spec
+        if isinstance(param, string_types):
+            param = MatchSpec(param)
+        if isinstance(param, MatchSpec):
+            return (prefix_rec for prefix_rec in self.iter_records()
+                    if param.match(prefix_rec))
+        else:
+            # assume isinstance(param, PackageRef)
+            return (prefix_rec for prefix_rec in self.iter_records() if prefix_rec == param)
+
```

Comment:
I'm surprised this `query` method didn't exist already.  In conda 4.4, I've intended `query()` to be an API method for conda's *three sources of information*.  i.e. `PackageCache`, `PrefixData`, and `SubdirData`.

Reply:
Maybe before we lock this API down completely, we should fix the inconsistency in the naming here.  So the class names would be `PackageCache`, `Prefix`, and `Subdir`.  That's problematic though, because we typically use lower-case variable names for class instances.  But we also use `prefix` and `subdir` all over the place just as strings that point to prefix locations or the string value of a subdir.

So, I guess we could have `PackageCacheData`, `PrefixData`, and `SubdirData`.  But that's kinda yuck too.

Reply:
The name clash with instances and other `prefix`/`subdir` should be avoided, yes, so +1 for `..Data` (and I usually weigh consistency/clarity a tad more than "yuck" 😉).

Reply:
Obviously I at first thought I wanted the `PackageCache`, `Prefix`, and `Subdir`.  As I was typing it out though, I convinced myself too that the latter is better.